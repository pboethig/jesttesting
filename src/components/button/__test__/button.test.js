import React from 'react';
import ReactDOM from 'react-dom';
import Button from '../button.js';
import {isTSAnyKeyword} from '@babel/types';
import {render, cleanup} from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer';

afterEach(cleanup);

it("renders without creashing", ()=>{
    const div = document.createElement("div");
    ReactDOM.render(<Button />, div);
});


it("renders button correctly", ()=>{
    const {getByTestId} = render( <Button label="click me"></Button>);
    expect(getByTestId('button')).toHaveTextContent("click me");
});


it("renders button correctly22", ()=>{
    const {getByTestId} = render( <Button label="save"></Button>);
    expect(getByTestId('button')).toHaveTextContent("save");
});

it("matches snapshots", ()=>{
   const tree = renderer.create(<Button label="save"></Button>).toJSON();

   expect(tree).toMatchSnapshot();
});

it("matches snapshots2", ()=>{
    const tree = renderer.create(<Button label="click me"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
});

it("matches snapshots3", ()=>{
    const component = renderer.create(<Button label="click me"></Button>).getInstance();

    let tree = component.dataFunction();

    expect(tree).toBe(10);
});

it("matches snapshots4", ()=>{
    const component = renderer.create(<Button label="click me"></Button>).getInstance();

    let tree = component.atestfunction();

    expect(tree).toBe(12);
});



