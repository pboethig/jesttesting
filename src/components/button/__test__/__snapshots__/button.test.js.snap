// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hi  
  save
</div>
`;

exports[`matches snapshots2 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hi  
  click me
</div>
`;

exports[`matches snapshots2 2`] = `
Button {
  "_reactInternalFiber": FiberNode {
    "_debugHookTypes": null,
    "_debugID": 15,
    "_debugIsCurrentlyTiming": false,
    "_debugNeedsRemount": false,
    "_debugOwner": null,
    "_debugSource": Object {
      "columnNumber": 34,
      "fileName": "/home/pbeothig/PhpstormProjects/testing/src/components/button/__test__/button.test.js",
      "lineNumber": 40,
    },
    "actualDuration": 0,
    "actualStartTime": -1,
    "alternate": null,
    "child": FiberNode {
      "_debugHookTypes": null,
      "_debugID": 16,
      "_debugIsCurrentlyTiming": false,
      "_debugNeedsRemount": false,
      "_debugOwner": [Circular],
      "_debugSource": Object {
        "columnNumber": 16,
        "fileName": "/home/pbeothig/PhpstormProjects/testing/src/components/button/button.js",
        "lineNumber": 16,
      },
      "actualDuration": 0,
      "actualStartTime": -1,
      "alternate": null,
      "child": FiberNode {
        "_debugHookTypes": null,
        "_debugID": 17,
        "_debugIsCurrentlyTiming": false,
        "_debugNeedsRemount": false,
        "_debugOwner": null,
        "_debugSource": null,
        "actualDuration": 0,
        "actualStartTime": -1,
        "alternate": null,
        "child": null,
        "childExpirationTime": 0,
        "dependencies": null,
        "effectTag": 0,
        "elementType": null,
        "expirationTime": 0,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": "Hi  ",
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": "Hi  ",
        "ref": null,
        "return": [Circular],
        "selfBaseDuration": 0,
        "sibling": FiberNode {
          "_debugHookTypes": null,
          "_debugID": 18,
          "_debugIsCurrentlyTiming": false,
          "_debugNeedsRemount": false,
          "_debugOwner": null,
          "_debugSource": null,
          "actualDuration": 0,
          "actualStartTime": -1,
          "alternate": null,
          "child": null,
          "childExpirationTime": 0,
          "dependencies": null,
          "effectTag": 0,
          "elementType": null,
          "expirationTime": 0,
          "firstEffect": null,
          "index": 1,
          "key": null,
          "lastEffect": null,
          "memoizedProps": "click me",
          "memoizedState": null,
          "mode": 0,
          "nextEffect": null,
          "pendingProps": "click me",
          "ref": null,
          "return": [Circular],
          "selfBaseDuration": 0,
          "sibling": null,
          "stateNode": Object {
            "isHidden": false,
            "tag": "TEXT",
            "text": "click me",
          },
          "tag": 6,
          "treeBaseDuration": 0,
          "type": null,
          "updateQueue": null,
        },
        "stateNode": Object {
          "isHidden": false,
          "tag": "TEXT",
          "text": "Hi  ",
        },
        "tag": 6,
        "treeBaseDuration": 0,
        "type": null,
        "updateQueue": null,
      },
      "childExpirationTime": 0,
      "dependencies": null,
      "effectTag": 0,
      "elementType": "div",
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {
        "children": Array [
          "Hi  ",
          "click me",
        ],
        "className": "button-style",
        "data-testid": "button",
      },
      "memoizedState": null,
      "mode": 0,
      "nextEffect": null,
      "pendingProps": Object {
        "children": Array [
          "Hi  ",
          "click me",
        ],
        "className": "button-style",
        "data-testid": "button",
      },
      "ref": null,
      "return": [Circular],
      "selfBaseDuration": 0,
      "sibling": null,
      "stateNode": Object {
        "children": Array [
          Object {
            "isHidden": false,
            "tag": "TEXT",
            "text": "Hi  ",
          },
          Object {
            "isHidden": false,
            "tag": "TEXT",
            "text": "click me",
          },
        ],
        "internalInstanceHandle": [Circular],
        "isHidden": false,
        "props": Object {
          "children": Array [
            "Hi  ",
            "click me",
          ],
          "className": "button-style",
          "data-testid": "button",
        },
        "rootContainerInstance": Object {
          "children": Array [
            [Circular],
          ],
          "createNodeMock": [Function],
          "tag": "CONTAINER",
        },
        "tag": "INSTANCE",
        "type": "div",
      },
      "tag": 5,
      "treeBaseDuration": 0,
      "type": "div",
      "updateQueue": null,
    },
    "childExpirationTime": 0,
    "dependencies": null,
    "effectTag": 1,
    "elementType": [Function],
    "expirationTime": 0,
    "firstEffect": null,
    "index": 0,
    "key": null,
    "lastEffect": null,
    "memoizedProps": Object {
      "label": "click me",
    },
    "memoizedState": Object {
      "label": "click me",
    },
    "mode": 0,
    "nextEffect": null,
    "pendingProps": Object {
      "label": "click me",
    },
    "ref": null,
    "return": FiberNode {
      "_debugHookTypes": null,
      "_debugID": 13,
      "_debugIsCurrentlyTiming": false,
      "_debugNeedsRemount": false,
      "_debugOwner": null,
      "_debugSource": null,
      "actualDuration": 0,
      "actualStartTime": -1,
      "alternate": FiberNode {
        "_debugHookTypes": null,
        "_debugID": 13,
        "_debugIsCurrentlyTiming": false,
        "_debugNeedsRemount": false,
        "_debugOwner": null,
        "_debugSource": null,
        "actualDuration": 0,
        "actualStartTime": -1,
        "alternate": [Circular],
        "child": null,
        "childExpirationTime": 0,
        "dependencies": null,
        "effectTag": 0,
        "elementType": null,
        "expirationTime": 1073741823,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": null,
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": null,
        "selfBaseDuration": 0,
        "sibling": null,
        "stateNode": FiberRootNode {
          "callbackExpirationTime": 0,
          "callbackNode": null,
          "callbackPriority": 90,
          "containerInfo": Object {
            "children": Array [
              Object {
                "children": Array [
                  Object {
                    "isHidden": false,
                    "tag": "TEXT",
                    "text": "Hi  ",
                  },
                  Object {
                    "isHidden": false,
                    "tag": "TEXT",
                    "text": "click me",
                  },
                ],
                "internalInstanceHandle": FiberNode {
                  "_debugHookTypes": null,
                  "_debugID": 16,
                  "_debugIsCurrentlyTiming": false,
                  "_debugNeedsRemount": false,
                  "_debugOwner": [Circular],
                  "_debugSource": Object {
                    "columnNumber": 16,
                    "fileName": "/home/pbeothig/PhpstormProjects/testing/src/components/button/button.js",
                    "lineNumber": 16,
                  },
                  "actualDuration": 0,
                  "actualStartTime": -1,
                  "alternate": null,
                  "child": FiberNode {
                    "_debugHookTypes": null,
                    "_debugID": 17,
                    "_debugIsCurrentlyTiming": false,
                    "_debugNeedsRemount": false,
                    "_debugOwner": null,
                    "_debugSource": null,
                    "actualDuration": 0,
                    "actualStartTime": -1,
                    "alternate": null,
                    "child": null,
                    "childExpirationTime": 0,
                    "dependencies": null,
                    "effectTag": 0,
                    "elementType": null,
                    "expirationTime": 0,
                    "firstEffect": null,
                    "index": 0,
                    "key": null,
                    "lastEffect": null,
                    "memoizedProps": "Hi  ",
                    "memoizedState": null,
                    "mode": 0,
                    "nextEffect": null,
                    "pendingProps": "Hi  ",
                    "ref": null,
                    "return": [Circular],
                    "selfBaseDuration": 0,
                    "sibling": FiberNode {
                      "_debugHookTypes": null,
                      "_debugID": 18,
                      "_debugIsCurrentlyTiming": false,
                      "_debugNeedsRemount": false,
                      "_debugOwner": null,
                      "_debugSource": null,
                      "actualDuration": 0,
                      "actualStartTime": -1,
                      "alternate": null,
                      "child": null,
                      "childExpirationTime": 0,
                      "dependencies": null,
                      "effectTag": 0,
                      "elementType": null,
                      "expirationTime": 0,
                      "firstEffect": null,
                      "index": 1,
                      "key": null,
                      "lastEffect": null,
                      "memoizedProps": "click me",
                      "memoizedState": null,
                      "mode": 0,
                      "nextEffect": null,
                      "pendingProps": "click me",
                      "ref": null,
                      "return": [Circular],
                      "selfBaseDuration": 0,
                      "sibling": null,
                      "stateNode": Object {
                        "isHidden": false,
                        "tag": "TEXT",
                        "text": "click me",
                      },
                      "tag": 6,
                      "treeBaseDuration": 0,
                      "type": null,
                      "updateQueue": null,
                    },
                    "stateNode": Object {
                      "isHidden": false,
                      "tag": "TEXT",
                      "text": "Hi  ",
                    },
                    "tag": 6,
                    "treeBaseDuration": 0,
                    "type": null,
                    "updateQueue": null,
                  },
                  "childExpirationTime": 0,
                  "dependencies": null,
                  "effectTag": 0,
                  "elementType": "div",
                  "expirationTime": 0,
                  "firstEffect": null,
                  "index": 0,
                  "key": null,
                  "lastEffect": null,
                  "memoizedProps": Object {
                    "children": Array [
                      "Hi  ",
                      "click me",
                    ],
                    "className": "button-style",
                    "data-testid": "button",
                  },
                  "memoizedState": null,
                  "mode": 0,
                  "nextEffect": null,
                  "pendingProps": Object {
                    "children": Array [
                      "Hi  ",
                      "click me",
                    ],
                    "className": "button-style",
                    "data-testid": "button",
                  },
                  "ref": null,
                  "return": [Circular],
                  "selfBaseDuration": 0,
                  "sibling": null,
                  "stateNode": [Circular],
                  "tag": 5,
                  "treeBaseDuration": 0,
                  "type": "div",
                  "updateQueue": null,
                },
                "isHidden": false,
                "props": Object {
                  "children": Array [
                    "Hi  ",
                    "click me",
                  ],
                  "className": "button-style",
                  "data-testid": "button",
                },
                "rootContainerInstance": [Circular],
                "tag": "INSTANCE",
                "type": "div",
              },
            ],
            "createNodeMock": [Function],
            "tag": "CONTAINER",
          },
          "context": Object {},
          "current": [Circular],
          "finishedExpirationTime": 0,
          "finishedWork": null,
          "firstPendingTime": 0,
          "firstSuspendedTime": 0,
          "hydrate": false,
          "interactionThreadID": 6,
          "lastExpiredTime": 0,
          "lastPingedTime": 0,
          "lastSuspendedTime": 0,
          "memoizedInteractions": Set {},
          "nextKnownPendingLevel": 0,
          "pendingChildren": null,
          "pendingContext": null,
          "pendingInteractionMap": Map {},
          "pingCache": null,
          "tag": 0,
          "timeoutHandle": -1,
        },
        "tag": 3,
        "treeBaseDuration": 0,
        "type": null,
        "updateQueue": Object {
          "baseQueue": Object {
            "callback": null,
            "expirationTime": 1073741823,
            "next": [Circular],
            "payload": Object {
              "element": <Button
                label="click me"
              />,
            },
            "priority": 97,
            "suspenseConfig": null,
            "tag": 0,
          },
          "baseState": null,
          "effects": null,
          "shared": Object {
            "pending": null,
          },
        },
      },
      "child": [Circular],
      "childExpirationTime": 0,
      "dependencies": null,
      "effectTag": 0,
      "elementType": null,
      "expirationTime": 0,
      "firstEffect": [Circular],
      "index": 0,
      "key": null,
      "lastEffect": [Circular],
      "memoizedProps": null,
      "memoizedState": Object {
        "element": <Button
          label="click me"
        />,
      },
      "mode": 0,
      "nextEffect": null,
      "pendingProps": null,
      "ref": null,
      "return": null,
      "selfBaseDuration": 0,
      "sibling": null,
      "stateNode": FiberRootNode {
        "callbackExpirationTime": 0,
        "callbackNode": null,
        "callbackPriority": 90,
        "containerInfo": Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "isHidden": false,
                  "tag": "TEXT",
                  "text": "Hi  ",
                },
                Object {
                  "isHidden": false,
                  "tag": "TEXT",
                  "text": "click me",
                },
              ],
              "internalInstanceHandle": FiberNode {
                "_debugHookTypes": null,
                "_debugID": 16,
                "_debugIsCurrentlyTiming": false,
                "_debugNeedsRemount": false,
                "_debugOwner": [Circular],
                "_debugSource": Object {
                  "columnNumber": 16,
                  "fileName": "/home/pbeothig/PhpstormProjects/testing/src/components/button/button.js",
                  "lineNumber": 16,
                },
                "actualDuration": 0,
                "actualStartTime": -1,
                "alternate": null,
                "child": FiberNode {
                  "_debugHookTypes": null,
                  "_debugID": 17,
                  "_debugIsCurrentlyTiming": false,
                  "_debugNeedsRemount": false,
                  "_debugOwner": null,
                  "_debugSource": null,
                  "actualDuration": 0,
                  "actualStartTime": -1,
                  "alternate": null,
                  "child": null,
                  "childExpirationTime": 0,
                  "dependencies": null,
                  "effectTag": 0,
                  "elementType": null,
                  "expirationTime": 0,
                  "firstEffect": null,
                  "index": 0,
                  "key": null,
                  "lastEffect": null,
                  "memoizedProps": "Hi  ",
                  "memoizedState": null,
                  "mode": 0,
                  "nextEffect": null,
                  "pendingProps": "Hi  ",
                  "ref": null,
                  "return": [Circular],
                  "selfBaseDuration": 0,
                  "sibling": FiberNode {
                    "_debugHookTypes": null,
                    "_debugID": 18,
                    "_debugIsCurrentlyTiming": false,
                    "_debugNeedsRemount": false,
                    "_debugOwner": null,
                    "_debugSource": null,
                    "actualDuration": 0,
                    "actualStartTime": -1,
                    "alternate": null,
                    "child": null,
                    "childExpirationTime": 0,
                    "dependencies": null,
                    "effectTag": 0,
                    "elementType": null,
                    "expirationTime": 0,
                    "firstEffect": null,
                    "index": 1,
                    "key": null,
                    "lastEffect": null,
                    "memoizedProps": "click me",
                    "memoizedState": null,
                    "mode": 0,
                    "nextEffect": null,
                    "pendingProps": "click me",
                    "ref": null,
                    "return": [Circular],
                    "selfBaseDuration": 0,
                    "sibling": null,
                    "stateNode": Object {
                      "isHidden": false,
                      "tag": "TEXT",
                      "text": "click me",
                    },
                    "tag": 6,
                    "treeBaseDuration": 0,
                    "type": null,
                    "updateQueue": null,
                  },
                  "stateNode": Object {
                    "isHidden": false,
                    "tag": "TEXT",
                    "text": "Hi  ",
                  },
                  "tag": 6,
                  "treeBaseDuration": 0,
                  "type": null,
                  "updateQueue": null,
                },
                "childExpirationTime": 0,
                "dependencies": null,
                "effectTag": 0,
                "elementType": "div",
                "expirationTime": 0,
                "firstEffect": null,
                "index": 0,
                "key": null,
                "lastEffect": null,
                "memoizedProps": Object {
                  "children": Array [
                    "Hi  ",
                    "click me",
                  ],
                  "className": "button-style",
                  "data-testid": "button",
                },
                "memoizedState": null,
                "mode": 0,
                "nextEffect": null,
                "pendingProps": Object {
                  "children": Array [
                    "Hi  ",
                    "click me",
                  ],
                  "className": "button-style",
                  "data-testid": "button",
                },
                "ref": null,
                "return": [Circular],
                "selfBaseDuration": 0,
                "sibling": null,
                "stateNode": [Circular],
                "tag": 5,
                "treeBaseDuration": 0,
                "type": "div",
                "updateQueue": null,
              },
              "isHidden": false,
              "props": Object {
                "children": Array [
                  "Hi  ",
                  "click me",
                ],
                "className": "button-style",
                "data-testid": "button",
              },
              "rootContainerInstance": [Circular],
              "tag": "INSTANCE",
              "type": "div",
            },
          ],
          "createNodeMock": [Function],
          "tag": "CONTAINER",
        },
        "context": Object {},
        "current": [Circular],
        "finishedExpirationTime": 0,
        "finishedWork": null,
        "firstPendingTime": 0,
        "firstSuspendedTime": 0,
        "hydrate": false,
        "interactionThreadID": 6,
        "lastExpiredTime": 0,
        "lastPingedTime": 0,
        "lastSuspendedTime": 0,
        "memoizedInteractions": Set {},
        "nextKnownPendingLevel": 0,
        "pendingChildren": null,
        "pendingContext": null,
        "pendingInteractionMap": Map {},
        "pingCache": null,
        "tag": 0,
        "timeoutHandle": -1,
      },
      "tag": 3,
      "treeBaseDuration": 0,
      "type": null,
      "updateQueue": Object {
        "baseQueue": null,
        "baseState": Object {
          "element": <Button
            label="click me"
          />,
        },
        "effects": null,
        "shared": Object {
          "pending": null,
        },
      },
    },
    "selfBaseDuration": 0,
    "sibling": null,
    "stateNode": [Circular],
    "tag": 1,
    "treeBaseDuration": 0,
    "type": [Function],
    "updateQueue": Object {
      "baseQueue": null,
      "baseState": Object {
        "label": "click me",
      },
      "effects": null,
      "shared": Object {
        "pending": null,
      },
    },
  },
  "_reactInternalInstance": Object {},
  "context": Object {},
  "props": Object {
    "label": "click me",
  },
  "refs": Object {},
  "state": Object {
    "label": "click me",
  },
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;
